patches/tModLoader/Terraria/ModLoader/ResourcesDisplayStyleLoader.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.GameContent.UI;
+
+namespace Terraria.ModLoader
+{
+	public class ResourcesDisplayStyleLoader
+	{
+		// There are no display set IDs, they're stored in their own list
+		// There are 3 of them
+		public const int VanillaDisplayStyleCount = 4;
+
+		public static int DisplayStyleCount => VanillaDisplayStyleCount + displayStyles.Count;
+		internal static readonly List<ModResourcesDisplayStyle> displayStyles = new List<ModResourcesDisplayStyle>();
+
+		internal static int Add(ModResourcesDisplayStyle displayStyle) {
+			if (ModNet.AllowVanillaClients)
+				throw new Exception("Adding display styles breaks vanilla client compatibility");
+
```

Comment:
Didn't notice this yesterday. This is a clientside thing. Doesn't break anything. ```suggestion
		internal static int Add(ModResourcesDisplayStyle displayStyle) {
```